[Chin Sek Yi] IP

ChinSekYi opened the pull request:
# Carly

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Carly frees your mind of having to remember things you need to do. 
It is:
- text-based
- fast and easy to learn
- ~~fast~~ ***super fast*** to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2425s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add in your tasks!!
4. then let it manage your tasks for you!! 😎✨

yo..and it's **FREE**!

Here are some features:
- [x] managing tasks like todos, deadlines and events
- [ ] managing reminders __(coming soon)__

---
If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```
public static void main(String[] args) {
    Carly carly = new Carly();
    try {
        carly.chat();
    } catch (Exception e) {
        System.out.println(e.getMessage());
    }
}
```

> [!NOTE]
> Stay tuned!

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

ChinSekYi committed:
add level-0

ChinSekYi committed:
Level-1

ChinSekYi committed:
add level-2

ChinSekYi committed:
add Level-3

ChinSekYi committed:
Level-4

ChinSekYi committed:
A-TextUiTesting

ChinSekYi committed:
add temp changes

ChinSekYi committed:
add Level-5

ChinSekYi committed:
add Level-6

ChinSekYi committed:
add A-Enums

ChinSekYi committed:
Add javadoc comments

ChinSekYi committed:
Add Level-7

ChinSekYi committed:
Add Level-8

ChinSekYi committed:
Update ui-test

ChinSekYi committed:
Add A-MoreOOP

ChinSekYi committed:
Add A-Packages

ChinSekYi committed:
Fix Ui bugs

ChinSekYi committed:
Merge branch 'add-gradle-support'

ChinSekYi committed:
Update build.gradle

ChinSekYi committed:
Update A-Gradle

ChinSekYi committed:
Add A-JUnit

ChinSekYi committed:
Add A-Jar

ChinSekYi committed:
Add A-JavaDoc

ChinSekYi committed:
Edit code format

ChinSekYi committed:
Add A-CodingStandard

ChinSekYi committed:
merge branch-A-CodingStandard to master

ChinSekYi committed:
Add Level-9

ChinSekYi committed:
Edit gitignore

ChinSekYi committed:
configure checkstyle

ChinSekYi committed:
Set up GUI

ChinSekYi committed:
Add Level-10

ChinSekYi committed:
Fix GUI bugs, disable checkstyle

ChinSekYi committed:
Configure GUI

ChinSekYi committed:
Add GUI elements

keerthigkaarthik commented on src/main/java/carly/Carly.java:
```diff
@@ -0,0 +1,217 @@
+                break;
+            case FIND:
+                try {
+                    this.taskList.find(taskDescription);
+                    break;
+                } catch (CarlyException e) {
+                    System.out.println(e.getMessage());
+                }
+                break;
+            case TODO:
+                try {
+                    this.taskList.addToDo(taskDescription);
+                } catch (CarlyException e) {
+                    System.out.println(e.getMessage());
+                }
+                break;
+            case DEADLINE:
+                try {
+                    this.taskList.addDeadLine(taskDescription);
+                } catch (CarlyException e) {
+                    System.out.println(e.getMessage());
+                }
+                break;
+            case EVENT:
+                try {
+                    this.taskList.addEvent(taskDescription);
+                } catch (CarlyException e) {
+                    System.out.println(e.getMessage());
+                }
+                break;
+            default:
+                Ui.printOutput("Oops, please print a valid command! ");
+            }
+            try {
+                listStorage.savesFile(this.taskList);
+            } catch (IOException | CarlyException e) {
+                System.out.println(e.getMessage());
+            }
+        }
+    }
```
Perhaps you could consider moving this into your Parser class? It could make your code look a bit neater.

keerthigkaarthik commented on src/main/java/carly/Carly.java:
```diff
@@ -0,0 +1,217 @@
+                try {
+                    response = this.taskList.find(taskDescription);
+                    break;
+                } catch (CarlyException e) {
+                    response = e.getMessage();
+                }
+                break;
+            case TODO:
+                try {
+                    response = this.taskList.addToDo(taskDescription);
+                } catch (CarlyException e) {
+                    response = e.getMessage();
+                }
+                break;
+            case DEADLINE:
+                try {
+                    response = this.taskList.addDeadLine(taskDescription);
+                } catch (CarlyException e) {
+                    response = e.getMessage();
+                }
+                break;
+            case EVENT:
+                try {
+                    response = this.taskList.addEvent(taskDescription);
+                } catch (CarlyException e) {
+                    response = e.getMessage();
+                }
+                break;
+            default:
+                response = "Oops, what are you trying to say again?";
+            }
+
+            listStorage.savesFile(this.taskList);
+
+        } catch (IOException | CarlyException e) {
+            response = e.getMessage();
+        }
+
+        return response;
+    }
```
It seems like you might not be using this method anymore in favour of the chat method. You may want to consider removing this to clean up this file.

keerthigkaarthik commented on src/main/java/carly/tasks/Event.java:
```diff
@@ -0,0 +1,50 @@
+    private static final String TASK_TYPE = "E";
+
+    /** The end time of the event. */
+    private final String startTime;
+
+    /** The end time of the event. */
+    private final String endTime;
```
Maybe you could copy paste some of the date time object methods from your deadline class to add this functionality for your event class as well.

keerthigkaarthik reviewed (commented):
Nicely done!

bryanckh commented on src/main/java/carly/utils/Parser.java:
```diff
@@ -0,0 +1,70 @@
+     * @return the corresponding enum value.
+     * @throws CarlyUnknownIInputException if the input action does not match any known command.
+     */
+    public Command getCommand() throws CarlyUnknownIInputException {
```
Nice handling of exceptions!

bryanckh commented on src/main/java/carly/utils/TaskList.java:
```diff
@@ -0,0 +1,235 @@
+     * @param taskDescription The description, start time, and end time of the task.
+     * @throws CarlyMissingDateTimeException If the task description, start time, or end time is missing.
+     */
+    public String addEvent(String taskDescription) throws CarlyMissingDateTimeException {
+        try {
+            String[] taskTimeParts = taskDescription.split(" /from ");
+            String task = taskTimeParts[0];
+            String timeParts = taskTimeParts[1];
+            String[] startEndTimeParts = timeParts.split(" /to ");
+            String startTime = startEndTimeParts[0];
+            String endTime = startEndTimeParts[1];
+
+            Event t = new Event(task, startTime, endTime);
+            this.taskList.add(t);
+            String msg = "Got it. I've added this task:\n" + TWO_INDENT + t;
+            return msg + "\n" + taskListSize();
+        } catch (ArrayIndexOutOfBoundsException e) {
+            throw new CarlyMissingDateTimeException("\"/from\" or \"/to\" command");
+        }
+    }
```
Good naming of methods and variables, well done!

bryanckh commented on src/main/java/toComplete_whenHaveTime.txt:
```diff
@@ -0,0 +1,40 @@
+- exceptions: event /from /to (empty description)
+- GUI: when bye is entered, carly should return bye before exiting.
+
+/*String logo = " ,-----.              ,--.          \n"
+                + "'  .--./,--,--.,--.--.|  |,--. ,--. \n"
+                + "|  |   ' ,-.  ||  .--'|  | \\  '  / \n"
+                + "'  '--'\\ '-'  ||  |   |  |  \\   ' \n"
+                + " `-----'`--`--'`--'   `--'.-'  /    \n"
+                + "                          `---'  ";
```
Nice logo!

bryanckh reviewed (commented):
Overall, the code was easy to read and the documentation was clear, well done!

ChinSekYi committed:
Fix master branch

ChinSekYi committed:
Add A-Assertions

ChinSekYi committed:
Improve code quality

- Consolidated mark and unmark methods into a shared helper function to reduce code duplication.- Added a parseTaskNumber method for cleaner task number parsing and better error handling.

ChinSekYi committed:
Merge pull request #3 from ChinSekYi/branch-A-CodeQuality

Improve code quality

ChinSekYi committed:
Edit java stream for printing tasklist

A java stream is used to loop through the tasklist to print out each tasks. In addition, each task printed out follows a particular string format.

ChinSekYi committed:
Merge pull request #4 from ChinSekYi/branch-A-Streams

Edit java stream for printing tasklist

ChinSekYi committed:
Add A-CI

ChinSekYi committed:
Update README.md

ChinSekYi committed:
Add sort extension

Sort command is added so that users are able to see a list of their tasks that are arranged based on the Deadline task duedate.

A sort function is therefore developed to deal with the sorting of tasks in the tasklist.

ChinSekYi committed:
Merge pull request #5 from ChinSekYi/branch-C-Sort

Add sort extension

ChinSekYi committed:
Improve code quality

ChinSekYi committed:
Improve code quality

ChinSekYi committed:
Implement FuzzyMatch

ChinSekYi committed:
Refactor ui

ChinSekYi committed:
Improve code quality

ChinSekYi committed:
Improve code quality

There are some methods that violates the code quality conventions.

Refactoring the code improves the code quality.

ChinSekYi committed:
Add apache commons text jar dependencies

ChinSekYi committed:
Improve GUI

ChinSekYi committed:
Edit GUI

ChinSekYi committed:
Add personality

ChinSekYi committed:
Update GUI

ChinSekYi committed:
Add User Guide

ChinSekYi committed:
Add User Guide

ChinSekYi committed:
Add User Guide

ChinSekYi committed:
Add User Guide

ChinSekYi committed:
Delete files

ChinSekYi committed:
Update README.md

ChinSekYi committed:
Delete Audio

ChinSekYi committed:
Add changes in CarlyList
